nvme_gw_server.py

```diff
@@ -0,0 +1,295 @@
+#
+#  Copyright (c) 2021 International Business Machines
+#  All rights reserved.
+#
+#  SPDX-License-Identifier: LGPL-3.0-or-later
+#
+#  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
+#
+
+import os
+import sys
+import subprocess
+import grpc
+from concurrent import futures
+from distutils import util
+import nvme_gw_pb2_grpc as pb2_grpc
+import nvme_gw_pb2 as pb2
+import nvme_gw_config
+import argparse
+
+
+class GWService(pb2_grpc.NVMEGatewayServicer):
+
+    def __init__(self, nvme_config):
+
+        self.logger = nvme_config.logger
+        self.nvme_config = nvme_config
+
+    def start_spdk(self):
+
+        spdk_path = self.nvme_config.get("config", "spdk_path")
+        sys.path.append(spdk_path)
+        self.logger.info(f"SPDK PATH: {spdk_path}")
+
+        import spdk.scripts.rpc as spdk_rpc
+
+        self.spdk_rpc = spdk_rpc
+        spdk_tgt = self.nvme_config.get("config", "spdk_tgt")
+        spdk_cmd = os.path.join(spdk_path, spdk_tgt)
+        cmd = [spdk_cmd]
+        cmd += ["all"]
+        cmd += ["-u"]
+        self.logger.info(f"Starting {' '.join(cmd)}")
+
+        try:
+            subprocess.Popen(cmd,
+                             stderr=subprocess.PIPE,
+                             stdout=subprocess.PIPE)
+
+        except Exception as ex:
+            self.logger.error(f"Unable start SPDK {ex}. Exiting!")
+            raise
+
+        spdk_server_addr = self.nvme_config.get("spdk", "spdk_server_addr")
+        spdk_port = self.nvme_config.get("spdk", "spdk_port")
+        timeout = self.nvme_config.getfloat("spdk", "timeout")
+        log_level = self.nvme_config.get("spdk", "log_level")
+        conn_retries = self.nvme_config.getint("spdk", "conn_retries")
+
+        self.logger.info(
+            f"Attempting to initialize the SPDK framework. server_addr: {spdk_server_addr},  port: {spdk_port}, conn_retry: {conn_retries}, timeout: {timeout}"
+        )
+
+        try:
+            self.client = self.spdk_rpc.client.JSONRPCClient(
+                spdk_server_addr,
+                spdk_port,
+                timeout,
+                log_level=log_level,
+                conn_retries=conn_retries,
+            )
+        except Exception as ex:
+            self.logger.error(f"Unable Initialize SPDK {ex}")
+            raise
+        return
+
+    def bdev_rbd_create(self, request, context):
+        # Create bdev from a given RBD image
+        message = request
+        self.logger.info({
+            f"Recieved: {message.ceph_pool_name}, {message.rbd_name}, {message.block_size} ",
```

Comment:
```suggestion
            f"Received: {message.ceph_pool_name}, {message.rbd_name}, {message.block_size}",
```
here and in other request handlers

Reply:
A quick grep still turns up three instances of "Recieved".